tests/lib/prepare.sh

```diff
@@ -505,26 +507,19 @@ prepare_all_snap() {
         fi
     done
 
+    echo "Ensure rsync is available"
+    if ! which rsync; then
+        snap install --devmode test-snapd-rsync
+        snap alias test-snapd-rsync.rsync rsync
+    fi
+
     disable_refreshes
     setup_systemd_snapd_overrides
 
     # Snapshot the fresh state (including boot/bootenv)
-    if [ ! -f "$SPREAD_PATH/snapd-state.tar.gz" ]; then
-        # we need to ensure that we also restore the boot environment
-        # fully for tests that break it
-        BOOT=""
-        if ls /boot/uboot/*; then
-            BOOT=/boot/uboot/
-        elif ls /boot/grub/*; then
-            BOOT=/boot/grub/
-        else
-            echo "Cannot determine bootdir in /boot:"
-            ls /boot
-            exit 1
-        fi
-
+    if [ ! -d $SNAPD_STATE_PATH ]; then
         systemctl stop snapd.service snapd.socket
-        tar cf "$SPREAD_PATH/snapd-state.tar.gz" /var/lib/snapd $BOOT /etc/systemd/system/snap-*core*.mount
+        save_snapd_state
```

Comment:
Why is the function called without an argument here and with an argument above? (in prepare_classic)

Reply:
With the last change, there is not any parameters sent to save_snapd_state anymore